### PR TITLE
Move the JSONObject structure from dl package to json and rename it to json.Object

### PR DESCRIPTION
**Pull request by supl**
Since we rename our module to `github.com/scalar-labs/scalardl-go-client-sdk/v3`.
The original package naming `dl` should be removed.
The only structure defined in the `dl` package is the JSONObject structure.
This PR moves it to the `json` package for better readability.

### CODE REVIEW

File: jsonobject/jsonobject.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package dl
+package jsonobject
```
Comment (supl) [inline jsonobject/jsonobject.go]: When json_object.go was in the root folder, we can assign it with any package name we want. (it is out of the Go source code structural package rule)

After this import statement (tell the importer to search the root folder)
```
import "github.com/scalar-labs/scalardl-go-client-sdk/v3"
```

it doesn't matter what your package name is, the importer can find it.

For example, if you have a Go file in the root folder defined with package foo.

```
package foo

struct Foo {
}
```

then, you can import it with
```
import "github.com/scalar-labs/scalardl-go-client-sdk/v3"

foo.Foo{
}
```

if your change the package name to
```
package bar

struct Foo {
}
```

you can import it with
```
import "github.com/scalar-labs/scalardl-go-client-sdk/v3"

bar.Foo{
}
```

This `feature` makes things a bit unclear than just
```
import "github.com/scalar-labs/scalardl-go-client-sdk/v3/foo"

foo.Foo{
}
```

This change is to make it more clear.

Comment (feeblefakie) [inline jsonobject/jsonobject.go]: Would it be better if it is `json`?
Other packages are like `client`, `crypto`, `ledger,`  and `rpc`, so `jsonobject` looks not consistent a bit to me.

More importantly, I thought this JSON file is intended for general JSON serialization and deserialization, but it's doing something for `nonce`, which is contract argument specific.
I think we should separate them. (maybe, withNonce should be done outside of the file?)

**Review by feeblefakie — COMMENTED** [review]
Thank you! Left some comments.

File: jsonobject/jsonobject.go (hunk 1 above)
Comment (supl) [reply jsonobject/jsonobject.go]: @feeblefakie 

Thank you for your feedback.

> Would it be better if it is json?
> Other packages are like client, crypto, ledger, and rpc, so jsonobject looks not consistent a bit to me.

Understood.
I revised the package name to `json` in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/18/commits/74c5845765ab88dc1f7939ba6bc1388a26da428a

And, according to Go linter's recommendation,
I renamed the structure `JSONObject` to `Object` in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/18/commits/661fc15fdf64f200e7a4e17908b2f7590d735087
(Go linter considers that `json.JSONObject` stutters and it recommends just `json.Object`)

> More importantly, I thought this JSON file is intended for general JSON serialization and deserialization, but it's doing something for nonce, which is contract argument specific.
I think we should separate them. (maybe, withNonce should be done outside of the file?)

I see. I removed the WithNonce function in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/18/commits/f6c8bd34e604714ecce50122f427317914b379a5
This part will be handled later in the `client/service` implementation directly.

Please take a look again.
Thank you!

File: json/object.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package json
+
+import (
+	ej "encoding/json"
+	"reflect"
+)
+
+// Object is a general structure to represent JavaScript objects.
```
Comment (feeblefakie) [inline json/object.go]:
```suggestion
// Object is a general structure to represent JSON objects.
```
I think `represent JavaScript object` sounds misleading.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Left one minor suggestion.

File: json/object.go (hunk 2 above)
Comment (supl) [reply json/object.go]: Thank you!

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**